agent/api/appmesh/appmesh_test.go

```diff
@@ -0,0 +1,173 @@
+// +build unit
+
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package appmesh
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/acs/model/ecsacs"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+const (
+	mockEgressIgnoredIP1   = "128.0.0.1"
+	mockEgressIgnoredIP2   = "171.1.3.24"
+	mockAppPort1           = "8000"
+	mockAppPort2           = "8001"
+	mockEgressIgnoredPort1 = "13000"
+	mockEgressIgnoredPort2 = "13001"
+	mockIgnoredUID         = "1337"
+	mockIgnoredGID         = "2339"
+	mockProxyIngressPort   = "9000"
+	mockProxyEgressPort    = "9001"
+	mockAppPorts           = mockAppPort1 + splitter + mockAppPort2
+	mockEgressIgnoredIPs   = mockEgressIgnoredIP1 + splitter + mockEgressIgnoredIP2
+	mockEgressIgnoredPorts = mockEgressIgnoredPort1 + splitter + mockEgressIgnoredPort2
+	mockContainerName      = "testEnvoyContainer"
+)
+
+func TestAppMeshFromACS(t *testing.T) {
+	testProxyConfig := prepareProxyConfig()
+
+	appMesh, err := AppMeshFromACS(&testProxyConfig)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, appMesh)
+	assert.True(t, appMesh.AppMeshEnabled)
+	assert.Equal(t, mockIgnoredUID, appMesh.IgnoredUID)
```

Comment:
Can u add test case to check AppMesh can be created successfully if optional parameters are missed?

Reply:
these tests were already included in this cr. but discussed offline that we don't need to validate anymore. will remove them